setuptools/__init__.pyi

```diff
@@ -0,0 +1,44 @@
+from typing import Any, Mapping, Optional, Sequence, Union
+
+import setuptools
+
+
+def setup(
+    name: str,
+    version: Optional[str] = None,
```

Comment:
I'm a little concerned about the "default" value of None. Will this signal to the consumer that passing `None` is the same as not passing the argument? I should think so, but the code expects the parameters to be absent ([example](https://github.com/pypa/setuptools/blob/b6bbe236ed0689f50b5148f1172510b975687e62/setuptools/_distutils/core.py#L100-L103)). Perhaps what's needed is keyword-only parameters with no defaults? e.g.:

```suggestion
    *,
    version: str,
```

Reply:
Hm, that is a good point. I think making them keyword-only parameters without defaults will be an issue though because `mypy` will warn the user about the missing argument (e.g. `error: Missing named argument "version" for "setup"  [call-arg]`). Note that the stub files are only read by `mypy` therefore this doesn't change the way `setup()` otherwise works and I think from the current documentation it should be clear how `setup()` should be used.

The point about making the arguments keyword-only is valid though, I'll change that. Somehow the ending ellipsis went missing too during the process so I'll add that as well.

Reply:
I've just found out in the [typeshed coding style guide](https://github.com/python/typeshed/blob/master/CONTRIBUTING.md#stub-file-coding-style) that an ellipsis is generally preferred for arguments with a default, therefore I changed `None` to `...` everywhere. This should also solve the issue with the misleading `None` showing up as the default.